test/sdn_migration/sdn_migration_test.go

```diff
@@ -181,11 +181,75 @@ var _ = Describe("SDN migration", Ordered, func() {
 		Expect(err).ShouldNot(HaveOccurred(), "Rosa Cluster failed to patch network")
 	})
 	It("rosa cluster has no critical alerts firing post sdn to ovn migration", postMigrationCheck, func(ctx context.Context) {
-		err := osdClusterReadyHealthCheck(ctx, testRosaCluster.client, "post-upgrade", testRosaCluster.reportDir)
+		err := cluterOperatorHealthCheck(ctx, testRosaCluster.client, logger)
+		err = osdClusterReadyHealthCheck(ctx, testRosaCluster.client, "post-upgrade", testRosaCluster.reportDir)
 		Expect(err).ShouldNot(HaveOccurred(), "osd-cluster-ready health check job failed post upgrade")
 	})
 })
 
+func cluterOperatorHealthCheck(ctx context.Context, clusterClient *openshiftclient.Client, logger logr.Logger) error {
+	var (
+		err    error
+		coList configv1.ClusterOperatorList
+		state  string
+	)
+
+	for i := 1; i <= upgradeMaxAttempts; i++ {
+		state = "healthy"
+		err = clusterClient.List(ctx, &coList)
+		if err != nil {
+			return fmt.Errorf("failed to get cluster operators: %v", err)
+		}
+
+		//Iterate over the list of ClusterOperators
+		if coList.Items == nil {
+			logger.Info("Failed to find cluster operators")
+		} else {
+			for _, co := range coList.Items {
+				// Check if the "Progressing" condition exists and is set to "false"
+				progressingCondition := findConditionByType(co.Status.Conditions, "Progressing")
+				availableCondition := findConditionByType(co.Status.Conditions, "Available")
+				if progressingCondition != nil && progressingCondition.Status == "True" || availableCondition != nil && availableCondition.Status == "False" {
+					logger.Info("waiting for cluster operators to be healthy")
+					state = "unhealthy"
+					break
+
```

Comment:
```suggestion
```
formatting

(it may help to run https://github.com/mvdan/gofumpt against your file to format it and fix a few of these issues)